pkg/userdata/ubuntu/userdata.go

```diff
@@ -353,29 +241,129 @@ write_files:
+    ExecStart=/usr/local/bin/kubelet \
+        --config=/var/lib/kubelet/config.yaml \
+        --container-runtime=remote \
+        --container-runtime-endpoint=unix:///var/run/containerd/containerd.sock \
+        --runtime-cgroups=/system.slice/containerd.service \
+        --runtime-request-timeout=15m \
+        --allow-privileged=true \
+        --cni-bin-dir=/opt/cni/bin \
+        --cni-conf-dir=/etc/cni/net.d \
+        --hostname-override={{ .MachineSpec.Name }} \
+        --network-plugin=cni \
+        {{- if .CloudProvider }}
+        --cloud-provider={{ .CloudProvider }} \
+        --cloud-config=/etc/kubernetes/cloud-config \
+        {{- end }}
+        --cert-dir=/etc/kubernetes/ \
+        --pod-manifest-path=/etc/kubernetes/manifests \
+        --kubeconfig=/etc/kubernetes/kubeconfig \
+        --bootstrap-kubeconfig=/etc/kubernetes/bootstrap.kubeconfig 
+
+    Restart=on-failure
+    RestartSec=5
+
+{{ if ne .CloudConfig "" }}
+- path: "/etc/kubernetes/cloud-config"
+  content: |
+{{ .CloudConfig | indent 4 }}
+{{- end }}
+
+- path: "/etc/kubernetes/bootstrap.kubeconfig"
+  content: |
+{{ .Kubeconfig | indent 4 }}
+
+- path: "/etc/kubernetes/pki/ca.crt"
+  content: |
+{{ .KubernetesCACert | indent 4 }}
+
+- path: "/var/lib/kubelet/config.yaml"
+  content: |
+    address: 0.0.0.0
+    apiVersion: kubelet.config.k8s.io/v1beta1
+    authentication:
+      anonymous:
+        enabled: false
+      webhook:
+        cacheTTL: 2m0s
+        enabled: true
+      x509:
+        clientCAFile: /etc/kubernetes/pki/ca.crt
+    authorization:
+      mode: Webhook
+      webhook:
+        cacheAuthorizedTTL: 5m0s
+        cacheUnauthorizedTTL: 30s
+    cgroupDriver: cgroupfs
+    cgroupsPerQOS: true
+    clusterDNS:
+    {{- range .ClusterDNSIPs }}
+    - {{ . }}
+    {{- end }}
+    clusterDomain: cluster.local
+    containerLogMaxFiles: 5
+    containerLogMaxSize: 10Mi
+    contentType: application/vnd.kubernetes.protobuf
+    cpuCFSQuota: true
+    cpuManagerPolicy: none
+    cpuManagerReconcilePeriod: 10s
+    enableControllerAttachDetach: true
+    enableDebuggingHandlers: true
+    enforceNodeAllocatable:
+    - pods
+    eventBurst: 10
+    eventRecordQPS: 5
+    evictionHard:
+      imagefs.available: 15%
+      memory.available: 100Mi
+      nodefs.available: 10%
+      nodefs.inodesFree: 5%
+    evictionPressureTransitionPeriod: 5m0s
+    failSwapOn: true
+    fileCheckFrequency: 20s
+    hairpinMode: promiscuous-bridge
+    healthzBindAddress: 127.0.0.1
+    healthzPort: 10248
+    httpCheckFrequency: 20s
+    imageGCHighThresholdPercent: 85
+    imageGCLowThresholdPercent: 80
+    imageMinimumGCAge: 2m0s
+    iptablesDropBit: 15
+    iptablesMasqueradeBit: 14
+    kind: KubeletConfiguration
+    kubeAPIBurst: 10
+    kubeAPIQPS: 5
+    makeIPTablesUtilChains: true
+    maxOpenFiles: 1000000
+    maxPods: 110
+    nodeStatusUpdateFrequency: 10s
+    oomScoreAdj: -999
+    podPidsLimit: -1
```

Comment:
I am fine with having this a high number, but allowing forkbombs is probably a little over the top ;)